src/classes/BDI_MappingServiceAdvanced.cls

```diff
@@ -102,21 +102,42 @@ public with sharing class BDI_MappingServiceAdvanced implements BDI_MappingServi
         if (fieldMappingInstance == null) {
             fieldMappingInstance = new BDI_MappingServiceAdvanced();
             fieldMappingInstance.setFieldMappingSetName(null);
-            fieldMappingInstance.retrieveCustMetadata();
+            fieldMappingInstance.retrieveCustMetadata(false);
+        }
+        return fieldMappingInstance;
+    }
+
+    /*******************************************************************************************************
+    * @description return singleton instance of the class
+    *
+    * @param includeUnmappedFields: Determines whether or not to include simulated field mappings
+    * e.g. DataImport__c.Donation_Donor__c field or the various Data_Import_Object_Mapping__mdt
+    * Imported_Record_Field_Name__c fields.
+    */
+    public static BDI_MappingServiceAdvanced getInstance(Boolean includeUnmappedFields) {
+        if (fieldMappingInstance == null) {
+            fieldMappingInstance = new BDI_MappingServiceAdvanced();
+            fieldMappingInstance.setFieldMappingSetName(null);
+            fieldMappingInstance.retrieveCustMetadata(includeUnmappedFields);
         }
         return fieldMappingInstance;
     }
 
     /*******************************************************************************************************
     * @description return singleton instance of the class using specified field mapping set dev name
+    *
+    * @param includeUnmappedFields: Determines whether or not to include simulated field mappings
```

Comment:
Kind of pedantic, but can we include an @param for fieldMappingSetDevName as well?